src/components/picker/__tests__/index.spec.tsx

```diff
@@ -100,122 +97,185 @@ describe('Picker', () => {
+        await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeTruthy());
         driver.selectItem(countries[2].label);
-        expect(driver.isOpen()).toBeFalsy();
+        act(() => driver.dismissDialog());
+        await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeFalsy());
         expect(driver.getValue()).toEqual(countries[2].label);
       });
 
-      it('Should select multiple items', () => {
-        const driver = getDriver({useDialog: true, customPickerProps: {migrateDialog: true}, mode: 'MULTI'});
-        expect(driver.getValue()).toEqual(undefined);
+      it('Should select multiple items', async () => {
+        const driver = getDriver({...dialogProps, mode: 'MULTI'});
+        expect(driver.getValue()).toEqual('');
         expect(driver.isOpen()).toBeFalsy();
-        driver.open();
-        expect(driver.isOpen()).toBeTruthy();
+        act(() => driver.open());
+        await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeTruthy());
         driver.selectItem(countries[2].label);
         driver.selectItem(countries[4].label);
         driver.done();
+        await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeFalsy());
         expect(driver.getValue()).toEqual(`${countries[2].label}, ${countries[4].label}`);
       });
     });
 
-    it('Test onDismiss', () => {
+    it('Test onDismiss', async () => {
       const driver = getDriver({
-        useDialog: true,
-        customPickerProps: {migrateDialog: true},
-        pickerModalProps: {
-          onDismiss
+        ...dialogProps,
+        customPickerProps: {
+          dialogProps: {
+            onDismiss
+          }
         }
       });
-
-      expect(driver.isOpen()).toBeFalsy();
-      driver.open();
-      expect(driver.isOpen()).toBeTruthy();
-      driver.cancel();
       expect(driver.isOpen()).toBeFalsy();
-      expect(onDismiss).toHaveBeenCalledTimes(2); // TODO: this should be 1
+      act(() => driver.open());
+      await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeTruthy());
+      act(() => driver.dismissDialog());
+      await waitFor(() => expect(driver.dismissDialog(dialogProps.useDialog)).toBeFalsy());
+      expect(onDismiss).toHaveBeenCalledTimes(1);
     });
   });
 
-  // TODO: add tests for WheelPicker as well
+  // TODO: add tests for WheelPicker
+  // Note: the picker dialog should be opened and then the wheel picker should be rendered, this is the main issue with testing it for now
   // describe.skip('WheelPicker', () => {
   // });
+
+  //TODO: add more tests for different props
+  describe('Picker field types', () => {
+    it('should render a form picker', () => {
+      const driver = getDriver({fieldType: 'form'});
+      const textFieldDriver = driver.getInput();
+      act(() => expect(textFieldDriver.exists()).toBeTruthy());
+      expect(textFieldDriver.getStyle()).toEqual(expect.objectContaining({textAlign: 'left'}));
```

Comment:
What if the design changes?
Maybe the `testID`s should be different and verify this way; not sure.